compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -646,4 +647,22 @@ object TreeChecker {
       tp
     }
   }.apply(tp0)
+
+  /** Check that the tree only contains legal children trees */
+  object TreeNodeChecker extends untpd.TreeTraverser:
+    import untpd._
+    def run(tree: Tree)(using Context) = if !tree.isInstanceOf[TypeTree] then traverse(tree)
```

Comment:
I don't understand this condition `!tree.isInstanceOf[TypeTree]`. Isn't this the tree of the compilation unit?